[DOCS] Rule Execution Log (#1845)
elastic/security-docs: docs/detections/rules-ui-monitor.asciidoc

[role="xpack"]
== Monitor and troubleshoot rule executions

> The Rules page offers several ways to gain insight into how your detection rules are running:

Author: I think it could go either way in this context, but I intended it as singular, similar to "understanding" or "knowledge"

> The Rules page offers several ways to gain insight into how your detection rules are running:

Reviewer: Another option here could be: 

`The Rules page offers several ways to gain insight into the status of your detection rules.`

Author: ```suggestion
The Rules page offers several ways to gain insight into the status of your detection rules:
```

* <<rule-monitoring-tab, Rule Monitoring tab>> — The current state of all detection rules and their most recent executions. Go to the Rule Monitoring tab to get an overview of which rules are running, how long they're taking, and if they're having any trouble.

> * <<rule-execution-logs, Rule execution logs>> — Historical data for a single detection rule's executions over time. Consult the rule execution logs to understand how a particular rule is running and whether it's creating the alerts you expect.

Author: It's a little redundant, but intentionally: to emphasize the difference between the rule monitoring tab (multiple/all rules) and the rule execution logs (single rule).

Refer to the <<troubleshoot-signals>> section below for strategies on using these tools.

[float]
[[rule-monitoring-tab]]
=== Rule Monitoring tab

> To view a summary of all rule executions, including most recent failures and execution

Reviewer 2: ```suggestion
To view a summary of all rule executions, including the most recent failures and execution
```

> To view a summary of all rule executions, including most recent failures and execution

Reviewer 3: ```suggestion
To view a summary of all rule executions, including the most recent failures and execution
```

For detailed information on a rule, the alerts it generated, and associated errors, click on its name in the table. This also allows you to perform the same actions that are available on the <<rules-ui-management, **Rules** tab>>, such as modifying or deleting rules, activating or deactivating rules, exporting or importing rules, and duplicating prebuilt rules.

[float]
[[rule-execution-logs]]
=== Rule execution logs

Each detection rule execution is logged, including its success or failure, any warning or error messages, and how long it took to search for data, create alerts, and complete. This can help you troubleshoot a particular rule if it isn't behaving as expected (for example, if it isn't creating alerts or takes a long time to run).

To access a rule's execution log, go to *Detect* -> *Rules*, click the rule's name to open its details, then scroll down and select the *Rule execution logs* tab.

[role="screenshot"]
image::images/rule-execution-logs.png[Rule execution logs table]

You can hover over each column heading to display a tooltip about that column's data. Click a column heading to sort the table by that column.

> Use these controls to narrow down what's included in the logs table:

Reviewer: ```suggestion
Use these controls to filter what's included in the logs table:
```

For detailed information on a rule, the alerts it generated, and associated errors, click on a rule's name in the *All rules* table. This also allows you to perform the same actions that are available on the <<rules-ui-management, **Rules** tab>>, such as modifying or deleting rules, activating or deactivating rules, exporting or importing rules, and duplicating prebuilt rules.

[float]
[[rule-execution-logs]]
=== Rule execution logs

> Each detection rule keeps a log of every time it ran, its success or failure, any warning or error messages, and how long it took to search for data, create alerts, and complete its execution. This can help you troubleshoot a particular rule if it isn't behaving as expected (for example, if it isn't creating alerts or takes a long time to run).

Reviewer 3: ```suggestion
Each detection rule execution is logged, including its success or failure, any warning or error messages, and how long it took to search for data, create alerts, and complete. This can help you troubleshoot a particular rule if it isn't behaving as expected (for example, if it isn't creating alerts or takes a long time to run).
```
I could be wrong but my thinking here is that the rule itself does not keep a log, the platform does.

> Each detection rule keeps a log of every time it ran, its success or failure, any warning or error messages, and how long it took to search for data, create alerts, and complete its execution. This can help you troubleshoot a particular rule if it isn't behaving as expected (for example, if it isn't creating alerts or takes a long time to run).

Reviewer 2: Ignore this if Ben's edit is accurate.

> To access a rule's execution log, go to *Detect* -> *Rules*, click the rule's name to open the rule details, then scroll down and select the *Rule execution logs* tab.

Reviewer 3: ```suggestion
To access a rule's execution log, go to *Detect* -> *Rules*, click the rule's name to open its details, then scroll down and select the *Rule execution logs* tab.
```

[role="screenshot"]
image::images/rule-execution-logs.png[Rule execution logs table]

You can hover over each column heading to display a tooltip about that column's data. Click a column heading to sort the table by that column.

Use these controls to narrow down what's included in the logs table:

> * The *Status* drop-down filters the table by rule execution status: 

Author: @spong Please take a close look at these statuses to make sure I described them correctly. Thanks!

** *Succeeded*: The rule was able to complete its defined search. This doesn't necessarily mean it generated an alert, just that it ran without error.
** *Failed*: The rule encountered an error that prevented it from running. For example, a machine learning rule whose corresponding machine learning job wasn't running.
** *Partial failure*: Nothing prevented the rule from running, but it still might not have returned the expected results. For example, a custom query rule tried to search an index pattern that couldn't be found in {es}.

> * The date and time picker sets the time range of rule executions included in the table. This overrides the time range set at the top of the rule details page.

Reviewer 4: > This overrides the time range set at the top of the rule details page.

I had to re-read this one, but you're just meaning that it is separate/independent of the global time picker, correct? If so, 👍.

Author: @spong You're correct, but if you had to re-read this maybe it wasn't phrased well. I like your term "separate" so I'll try that:
```suggestion
* The date and time picker sets the time range of rule executions included in the table. This is separate from the global date and time picker at the top of the rule details page.
```

Reviewer 4: Sounds good - thanks Joe! 🙂
